docker/api/image.py

```diff
@@ -225,8 +225,19 @@ def import_image_from_image(self, image, repository=None, tag=None,
             image=image, repository=repository, tag=tag, changes=changes
         )
 
+    def set_auth_headers(self, registry, auth_config=None):
+        headers = {}
+        if auth_config is None:
+            header = auth.get_config_header(self, registry)
+            if header:
+                headers['X-Registry-Auth'] = header
+        else:
+            log.debug('Sending supplied auth config')
+            headers['X-Registry-Auth'] = auth.encode_header(auth_config)
+        return headers
+
     @utils.check_resource('image')
-    def inspect_image(self, image):
+    def inspect_image(self, image, auth_config=None):
```

Comment:
Instead of passing `auth_config` on all the client methods, you could just create a setter for it in the client. 
That would also help you with the commented unit test when checking the `inspect`.